File: frontend/src/api/quotes.ts
import request from './request'

export interface Quote {
  id: number
  content: string
  source: string
  created_at: string
}

export interface QuoteList {
  items: Quote[]
  total: number
}

// 获取语录列表
export const getQuotes = (params: { skip?: number; limit?: number }) => {
  return request.get<QuoteList>('/quotes', { params })
}

// 上传语录文件
export const uploadQuotes = (file: File) => {
  const formData = new FormData()
  formData.append('file', file)
  return request.post<Quote[]>('/quotes/upload', formData)
}

// 生成语录
export const generateQuotes = (prompt: string, count: number = 10) => {
  return request.post<Quote[]>('/quotes/generate', { prompt, count })
}

// 随机获取语录
export const getRandomQuotes = (count: number = 5) => {
  return request.get<Quote[]>('/quotes/random', { params: { count } })
} 

File: backend/app/routers/quotes.py
from fastapi import APIRouter, Depends, UploadFile, File, HTTPException
from sqlalchemy.orm import Session
from typing import List
from app.database import get_db
from app.services.kimi_service import kimi_service
from app.schemas.quote import QuoteCreate, QuoteResponse, QuoteGenerateRequest, QuoteList
from app.crud import quote as quote_crud

router = APIRouter()

@router.post("/upload", response_model=List[QuoteResponse])
async def upload_quotes(file: UploadFile = File(...), db: Session = Depends(get_db)):
    """上传TXT文件导入语录"""
    if not file.filename.endswith('.txt'):
        raise HTTPException(status_code=400, detail="只支持TXT文件")
    return await quote_crud.create_quotes_from_file(db, file)

@router.post("/generate", response_model=List[QuoteResponse])
async def generate_quotes(request: QuoteGenerateRequest, db: Session = Depends(get_db)):
    """使用Kimi AI生成语录"""
    quotes = await kimi_service.generate_quotes(request.prompt, request.count)
    return await quote_crud.create_quotes_from_list(db, quotes, f"Kimi AI - {request.prompt}")

@router.get("/", response_model=QuoteList)
async def get_quotes(skip: int = 0, limit: int = 100, db: Session = Depends(get_db)):
    """获取语录列表"""
    quotes, total = quote_crud.get_quotes(db, skip=skip, limit=limit)
    return {"items": quotes, "total": total}

@router.get("/random", response_model=List[QuoteResponse])
async def get_random_quotes(count: int = 5, db: Session = Depends(get_db)):
    """随机获取指定数量的语录"""
    return quote_crud.get_random_quotes(db, count)